tests/phpunit/unit/api/actions/beansResetAction.php

```diff
@@ -0,0 +1,199 @@
+ * @group   unit-tests
+ * @group   api
+ */
+class Tests_BeansResetAction extends Actions_Test_Case {
+
+	/**
+	 * Test beans_reset_action() should return false when the action is not registered.
+	 */
+	public function test_should_return_false_when_no_action() {
+
+		foreach ( static::$test_actions as $beans_id => $action ) {
+			$this->assertFalse( beans_reset_action( $beans_id ) );
+		}
+	}
+
+	/**
+	 * Test beans_reset_action() should reset the original action after it was "removed".
+	 */
+	public function test_should_reset_after_remove() {
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action ) {
+			// Check that the action is registered before we start.
+			$this->assertSame( $action, _beans_get_action( $beans_id, 'added' ) );
+			$this->assertTrue( has_action( $action['hook'], $action['callback'] ) !== false );
+
+			// Now remove it.
+			beans_remove_action( $beans_id );
+
+			// Check that it did get removed.
+			$this->assertSame( $action, _beans_get_action( $beans_id, 'removed' ) );
+			$this->assertFalse( has_action( $action['hook'], $action['callback'] ) );
+
+			// Let's reset the action.
+			$this->assertSame( $action, beans_reset_action( $beans_id ) );
+
+			// Check that it was reset.
+			$this->assertFalse( _beans_get_action( $beans_id, 'removed' ) );
+			$this->check_the_action( $beans_id, $action );
+		}
+	}
+
+	/**
+	 * Test beans_reset_action() should reset the original action's hook after it was "modified".
+	 */
+	public function test_should_reset_after_modifying_the_hook() {
+		$modified_action = array(
+			'hook' => 'foo',
+		);
+
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action ) {
+			// Check that the action is registered before we start.
+			$this->assertSame( $action, _beans_get_action( $beans_id, 'added' ) );
+			$this->assertTrue( has_action( $action['hook'], $action['callback'] ) !== false );
+
+			// Now modify the hook.
+			beans_modify_action_hook( $beans_id, $modified_action['hook'] );
+
+			// Check that the hook was modified.
+			$this->assertSame( $modified_action, _beans_get_action( $beans_id, 'modified' ) );
+			$this->assertFalse( has_action( $action['hook'], $action['callback'] ) );
+			$this->assertTrue( has_action( $modified_action['hook'], $action['callback'] ) !== false );
+
+			// Let's reset the action.
+			$this->assertSame( $action, beans_reset_action( $beans_id ) );
+
+			// Check that it was reset.
+			$this->assertFalse( _beans_get_action( $beans_id, 'modified' ) );
+			$this->check_the_action( $beans_id, $action );
+		}
+	}
+
+	/**
+	 * Test beans_reset_action() should reset the original action's callback after it was "modified".
+	 */
+	public function test_should_reset_after_modifying_the_callback() {
+		$modified_action = array(
+			'callback' => 'my_callback',
+		);
+
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action ) {
+			// Test the original callback.
```

Comment:
Should we stay consistent with the previous comments and say something like
`Check that the original callback is registered before we start.`?